docs: Create a document for the user’s manual steps (#1161)
platform-platform/monorepo: metrics/cli/docs/05_user_manual_steps.md

Here is a prompt instruction, which helps the user to configure the Sentry:

```text
Sentry configuration instruction:

To enable Sentry for the current deploying project, you should copy & paste a DSN from your Sentry account.

The following steps help to find a DSN for the Sentry account:
1. Visit the following link and authorize: https://sentry.io
2. Navigate to Settings -> Projects -> Client Keys (DSN)

Copy & paste your DSN here:
```

### Firebase auth configuration

Since the Metrics Project using Firebase Auth and the auth is disabled by default, we should provide a detailed instruction, 
which helps the user to configure a Firebase Auth with the [Email/Password](https://github.com/platform-platform/monorepo/blob/master/docs/09_firebase_deployment.md#firebase-email-and-password-sign-in-configuration) and [Google](https://github.com/platform-platform/monorepo/blob/master/docs/09_firebase_deployment.md#firebase-google-sign-in-configuration) auth providers during the Metrics deployment process using the Metrics CLI.

Here is a prompt instruction, which helps the user to configure Firebase auth:

```text

> Firebase auth configuration instruction:
> 
> To enable Firebase auth for the current deploying project, consider the following steps:

Reviewer: ```suggestion
To enable the Firebase Auth for the Metrics Web application, consider the following steps:

```

1. Follow the link and click a "Get started" button: https://console.firebase.google.com/project/project_id/authentication
2. Navigate to the Sign-in method's tab on the Firebase auth page or use the following link: https://console.firebase.google.com/project/project_id/authentication/providers
3. Enable an Email/Password provider.
4. Enable a Google provider.
5. On the Google provider page, open the Web SDK configuration tab, copy the Web client ID and paste it to the console

Paste your Web client Id here:
```
### Firebase Analytics configuration

> Since the Metrics Project using Firebase Analytics and the analytics is disabled by default, we should provide a detailed instruction,
> which helps the user to configure the Firebase analytics during the Metrics deployment process using the Metrics CLI.
> 
> Here is a prompt instruction, which helps the user to configure Firebase analytics easily:

Reviewer: ```suggestion
To make the Metrics Web application send the analytics data to the Firestore, the user should enable the Firestore Analytics service in the Firestore console. 

During the deployment process, we should display the following prompt explaining how to enable the Firestore analytics:
```

```text

> Firebase analytics configuration instruction:
> 
> To enable Firebase analytics for the current deploying project, consider the following steps:

Reviewer: ```suggestion
To enable Firebase analytics for the currently deploying project, follow the next steps:

```

1. Follow the link and click an "Enable Google Analytics" button: https://console.firebase.google.com/project/project_id/settings/integrations/analytics

> Press any key to continue:

Reviewer: ```suggestion
Once you are done, press any key to continue:
```

```
### Default region configuration

> Since the deployment of the Metrics project requires a GCloud project's region, we should provide a detailed instruction, 
> which helps the user to select the [region/zone](https://cloud.google.com/compute/docs/regions-zones) during the Metrics deployment process using the Metrics CLI.
> 
> Here is a prompt instruction, which helps the user to select a GCloud project's region:

Reviewer: ```suggestion
During the deployment process, the user should select the region the `GCloud services` will locate. Review the following prompt explaining the process of choosing the `GCloud` project region:
```

```text
Here is a list of available regions for gcloud project:

REGION                   SUPPORTS STANDARD  SUPPORTS FLEXIBLE
asia-east2               YES                YES
asia-northeast1          YES                YES
asia-northeast2          YES                YES
asia-northeast3          YES                YES
asia-south1              YES                YES
asia-southeast2          YES                YES
...............          ...                ...

Please enter the region name that suits you best:
```

> ### Initial Firestore Data configuration

Reviewer: ```suggestion
### Adding initial Firestore data
```

> Since the Metrics project controls the allowed email domains and features config using the Firestore database appropriate collections, 
> we should provide a detailed instruction, which helps the user to initialize the Firestore database with the mentioned collections' data during the Metrics deployment process using the Metrics CLI.
> 
> Here is a prompt instruction, which helps the user to initialize the Firestore database:

Reviewer: ```suggestion
Since the Metrics project controls the allowed email domains and features config using the Firestore database appropriate collections, we should provide a detailed instruction describing the initialization of the Firestore database with the initial data during the deployment process using the following prompt:
```

```text

> Initialize Firestore data instruction:
> 
> To initialize Firestore data for the current deploying project, consider the following steps:

Reviewer: ```suggestion
To initialize Firestore data for the current deploying project, consider the following steps:
```

1. Follow the link to open the Firestore database: https://console.firebase.google.com/project/project_id/firestore

> 2. Create a collection with the "allowed_email_domains" id.

Reviewer: ```suggestion
2. Create a collection with the "allowed_email_domains" identifier.
```

> 3. Create document(s) with the "domain" id without any fields.

Reviewer: It would be great to provide a link explaining the process of adding an allowed email domain in more detail. Also, the link explaining the structure of the `feature config` database document would help the user create it properly.

4. Create a collection with the "feature_config" id.
5. Create a document with the "feature_config" id.
6. Add the following boolean fields to the feature_config document:
   - "isDebugMenuEnabled" - Indicates whether the Debug Menu feature is enabled.
   - "isPasswordSignInOptionEnabled" - Indicates whether the Email and Password sign in option is enabled.

> Press any key to continue:

Reviewer: ```suggestion
Press any key to continue when you finish with the Firestore database initialization:
```

```
### Blaze account configuration

> Since some Firebase features, such as Firebase functions, are not available without a Blaze billing plan, we should provide a detailed instruction, which helps the user to configure the Blaze billing plan during the Metrics CLI deployment process.
> 
> Here is a prompt instruction, which helps the user to configure the Blaze billing plan:

Reviewer: ```suggestion
The Metrics Web application uses the Firebase functions that are not available in the Free Firebase plan, so the user should enable the Blaze billing plan. To simplify this process, we should add the following prompt containing detailed instructions on how to enable the Blaze plan: 

```

```text

> To configure Blaze billing plan for the current deploying project, consider the following steps:

Reviewer: ```suggestion
To configure the Blaze billing plan for the Firebase project, consider the following steps:
```

1. Follow the link and press continue in the open modal window: https://console.firebase.google.com/project/project_id/overview?purchaseBillingPlan=metered&billingContext=pricingBuyFlow
2. Select your country and press "Confirm"
3. Enter your customer info, choose payment method and press "Confirm Purchase"

> Press any key to continue:

Reviewer: ```suggestion
Press any key to continue once the billing plan configured: 
```

Consider the following prompt instructions helping to configure the Firebase Auth:

```text
To enable the Firebase Auth for the Metrics Web application, consider the following steps:

1. Follow the link and click a "Get started" button: https://console.firebase.google.com/project/project_id/authentication
2. Navigate to the 'Sign-in method' tab on the Firebase auth page or use the following link: https://console.firebase.google.com/project/project_id/authentication/providers
3. Enable an Email/Password provider by clicking on provider name and toggling the `Enable` switch.
4. Enable a Google provider by clicking on provider name and toggling the `Enable` switch.
5. On the 'Google' provider popup, open the 'Web SDK configuration' tab, copy the Web client ID and paste it to the console

Paste your Web client Id here:
```

### Sentry Configuration

Since the Metrics Web application uses Sentry, we should provide detailed instruction to help a user configure a Sentry release during the Metrics application deployment process.
Let's review prompts instructions that will help configure a new Sentry release.

> The first prompt for authorizing to the Sentry and retrieving an `Organization Slug`:

Reviewer: Would it be great to add a bit more info on what is `Organization Slug`, `Project Slug`, and `DSN` and why do we need them?

```text

> To enable Sentry for the Metrics Web application, consider the following steps:
> 
> The following steps help to find an `Organization Slug` for the Sentry account:

Reviewer: ```suggestion
The following steps help to find an `Organization Slug` for the Sentry account:
```

1. Visit the following link and authorize: https://sentry.io
2. Navigate to the Sentry's `Settings` tab
3. Navigate to the `General Settings` tab and copy an `Organization Slug`

Paste the `Organization Slug` here:
```

The prompt for retrieving a `Project Slug`:
```text
The following steps help to find a `Project Slug` for the Sentry account:

1. Navigate to the Sentry's `Settings` tab

> 2. Navigate tp the `Projects` tab and copy a required `Project Slug`

Reviewer: ```suggestion
2. Navigate to the `Projects` tab and copy a required `Project Slug`
```
Also, would it be great to give a link to users to simplify our steps? I think we can create a link since we already have an `organization slug`. What do you think?

Paste the `Project Slug` here:
```

The prompt for retrieving a `DSN`:
```text
The following steps help to find a `DSN` for the Sentry account:

1. Navigate to the Sentry's `Settings` tab
2. Navigate to the `Projects` tab
3. Select a required project 
4. Navigate to the Client Keys (DSN) tab and copy your DSN

Paste the DSN here:
```

The final prompt for retrieving a `Sentry release name`:
```text
The last thing required for the Sentry configuration is a release name. The Sentry web page will display the entered release name in the issues tags.

> While creating a release name, consider the following restrictions for it:  

Reviewer: ```suggestion
While creating a release name, consider it cannot: 
```
?
